
## Results and interpretation

The following directories will be created in the output directory after the pipeline has finished.

### `qc/`

#### QC

| Directory          | Description                                                                                                                                                                                                       |
|--------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `qc/fastqc/raw/`   | HTML files generated by [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) for read 1 and read2 **before** adapter trimming. Zip archives are in `zip/` directory.                              |
| `qc/fastqc/trim/`  | HTML files generated by [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) for read 1 and read2 **after** adapter trimming. Zip archives are in `zip/` directory.                               |
| `qc/fastq_screen/` | HTML files generated by [FastQ Screen](https://www.bioinformatics.babraham.ac.uk/projects/fastq_screen/) for read 1 and read2 **before** adapter trimming. Text output is in `txt/` directory.                    |
| `qc/cutadapt/`     | Log files generated by [cutadapt](http://cutadapt.readthedocs.io/en/stable/installation.html) containing adapter trimming information.                                                                            |
| `qc/multiqc/`      | HTML file generated by [MultiQC](http://multiqc.info/) to collate pipeline QC from FastQC, FastQ Screen, cutadapt, samtools flagstat, samtools idxstats, picard CollectMultipleMetrics and picard MarkDuplicates. |

### `align/`

#### Run-level results

| Directory               | Description                                                                                                                                                                                                                                              |
|-------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `align/`                | Filtered, coordinate sorted alignment files in [BAM](https://samtools.github.io/hts-specs/SAMv1.pdf) format at the run-level for each sample.                                                                                                            |
| `align/flagstat/`       | Throughout the pipeline various BAM files at the run-level will be generated before the final filtered BAM file is created. The associated [flagstat](http://www.htslib.org/doc/samtools.html) files for each of these will be placed in this directory. |
| `align/idxstats/`       | [idxstats](http://www.htslib.org/doc/samtools.html) files to determine the percentage of reads mapping to mitochondrial DNA.                                                                                                                             |
| `align/picard_metrics/` | Alignment QC files from picard [CollectMultipleMetrics](https://broadinstitute.github.io/picard/command-line-overview.html) and the metrics file from [MarkDuplicates](https://broadinstitute.github.io/picard/command-line-overview.html).              |
| `align/sysout/`         | Sysout files for various programs to aid in troubleshooting errors.                                                                                                                                                                                      |

#### Replicate-level results

| Directory                                                    | Description                                                                                                                                                                                       |
|--------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `align/mergeReplicate/`                                      | Replicate-level, merged, coordinate sorted BAM files after the re-marking and removal of duplicates.                                                                                              |
| `align/mergeReplicate/flagstat/`                             | Flagstat files associated with the merged BAM files before and after duplicate removal.                                                                                                           |
| `align/mergeReplicate/picard_metrics/`                       | Metrics file from MarkDuplicates.                                                                                                                                                                 |
| `align/mergeReplicate/sysout/`                               | Sysout files for various programs to aid in troubleshooting errors.                                                                                                                               |
| `align/mergeReplicate/bigwig/`                               | Normalised [bigWig](https://genome.ucsc.edu/goldenpath/help/bigWig.html) files scaled to 1 million mapped read pairs.                                                                             |
| `align/mergeReplicate/macs2/broadPeak/`                      | Peak sets per sample called in "broadPeak" mode with [MACS2](https://github.com/taoliu/MACS).                                                                                                     |
| `align/mergeReplicate/macs2/broadPeak/qc/`                   | Various QC plots per sample including number of peaks, fold-change distribution, [FRiP score](https://genome.cshlp.org/content/22/9/1813.full.pdf+html) and peak percentage across gene features. |
| `align/mergeReplicate/macs2/broadPeak/merged_peaks/`         | Merged "broadPeak" peaks across all samples.                                                                                                                                                      |
| `align/mergeReplicate/macs2/broadPeak/merged_peaks/deseq2`   | Differential binding analysis, PCA and clustering using reads quantified at the replicate-level for the "broadPeak" merged peak set.                                                              |
| `align/mergeReplicate/macs2/narrowPeak/`                     | Peak sets per sample called in "narrowPeak" mode with MACS2.                                                                                                                                      |
| `align/mergeReplicate/macs2/narrowPeak/qc/`                  | Various QC plots per sample including number of peaks, fold-change distribution, FRiP score and peak percentage across gene features.                                                             |
| `align/mergeReplicate/macs2/narrowPeak/merged_peaks`         | Merged "narrowPeak" peaks across all samples.                                                                                                                                                     |
| `align/mergeReplicate/macs2/narrowPeak/merged_peaks/deseq2/` | Differential binding analysis, PCA and clustering using reads quantified at the replicate-level for the "narrowPeak" merged peak set.                                                             |

#### Sample-level results

`align/mergeReplicate/` and `align/mergeSample/` have exactly the same directory structure. The only difference is that multiple libraries sequenced from the same sample will be merged in the `mergeReplicate/` directory, and all the replicates associated with a sample will be merged in the `mergeSample/` directory. The main reasoning behind merging the replicates at the sample-level was to improve the coverage in order to call more accurately defined and robust peaksets. This may have some downsides especially if there is high variability amongst the replicates. However, for the most part it seems to work quite well.

### `genome/`

Genome-specific files required by certain processes in the pipeline.

### `igv/`

An [IGV](https://software.broadinstitute.org/software/igv/) session file called `igv_session.xml` will be created at the end of the pipeline. This avoids having to load all the data individually into IGV for visualisation. Once installed, open IGV, go to `File > Open Session` and select the `igv_session.xml` file for loading.

File paths in the IGV session file will be set as absolute paths to the directory containing the results. If you prefer to load the data over the web you can just replace the relevant portion of the file path with a link in the session file.

The reference genome fasta file will be soft-linked to the `genome/` directory, and by default this will be set as the genome for the IGV session. If you prefer to use an in-built genome provided by IGV just change the file path to the name of the IGV genome e.g. mm10 or hg19.   

### `pipeline/`

Nextflow provides excellent functionality for generating various reports relevant to the running and execution of the pipeline. This will allow you to trouble-shoot errors with the running of the pipeline, and also provide you with other information such as launch commands, run times and resource usage. Default reports generated by the pipeline are `BABS-ATACSeqPE_report.html`, `BABS-ATACSeqPE_timeline.html`, `BABS-ATACSeqPE_trace.txt` and `BABS-ATACSeqPE_dag.dot`.

See [Nextflow Tracing & visualisation](https://www.nextflow.io/docs/latest/tracing.html)

<!---
Add information about specific files that will need to be used from results i.e. merged peaks, differential results
Add information about multiqc
Put small line below each section in align to explain files in that directory
alignments, peaks, differential analysis & QC
-->
